unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1849,28 +1859,27 @@ public class TypedVariable {
 
 
 
-public class ShouldSerializeContractResolver : DefaultContractResolver
-{
-   public static readonly ShouldSerializeContractResolver Instance = new ShouldSerializeContractResolver();
-
-   protected override JsonProperty CreateProperty( MemberInfo member,
-                                    MemberSerialization memberSerialization )
-   {
-      JsonProperty property = base.CreateProperty( member, memberSerialization );
-
-      // exclude these properties to make serialization match JsonUtility
-      if( property.DeclaringType == typeof(Vector3) &&
-            (property.PropertyName == "sqrMagnitude" || 
-            property.PropertyName == "magnitude"  ||
-            property.PropertyName == "normalized" 
-            ))
-      {
+public class ShouldSerializeContractResolver : DefaultContractResolver {
+    public static readonly ShouldSerializeContractResolver Instance = new ShouldSerializeContractResolver();
+
+    protected override JsonProperty CreateProperty(
+        MemberInfo member,
+        MemberSerialization memberSerialization
+    ) {
+       JsonProperty property = base.CreateProperty(member, memberSerialization);
+
+       // exclude these properties to make serialization match JsonUtility
+      if (property.DeclaringType == typeof(Vector3) &&
+          (property.PropertyName == "sqrMagnitude" ||
+          property.PropertyName == "magnitude"  ||
+          property.PropertyName == "normalized")
+      ) {
          property.ShouldSerialize = instance => { return false; };
          return property;
       } else {
           return base.CreateProperty(member, memberSerialization);
       }
-
+      return base.CreateProperty(member, memberSerialization);
```

Comment:
Can this return ever occur? Seems like the above if/else clause has returns in both branches.

Reply:
Yeah, that's true here. Was using it for debugging, since it was saying "Vector3.normalized" gave a self-recurring loop.